app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java

```diff
@@ -78,22 +78,31 @@ public ArrayList<Integer> getRemovedItems() {
         return removedItems;
     }
 
-    private MediaDetailPagerFragment() {}; // Constructor calls made to be explicit
-    @SuppressLint("ValidFragment")
-    public MediaDetailPagerFragment(Boolean editable, boolean isFeaturedImage) {
-        this.editable = editable;
-        this.isFeaturedImage = isFeaturedImage;
-        isFromFeaturedRootFragment = false;
-    }
 
-    @SuppressLint("ValidFragment")
-    public MediaDetailPagerFragment(Boolean editable, boolean isFeaturedImage, int position) {
-        this.editable = editable;
-        this.isFeaturedImage = isFeaturedImage;
-        isFromFeaturedRootFragment = true;
-        this.position = position;
+    /**
+     * Use this factory method to create a new instance of this fragment using the provided
+     * parameters.
+     * 
+     * This method will create a new instance of MediaDetailPagerFragment and the arguments will be
+     * saved to a bundle which will be later available in the {@link #onCreate(Bundle)}
+     * @param editable
+     * @param isFeaturedImage
+     * @return
+     */
+    public static MediaDetailPagerFragment newInstance(boolean editable, boolean isFeaturedImage) {
+        MediaDetailPagerFragment mediaDetailPagerFragment = new MediaDetailPagerFragment();
+        Bundle args = new Bundle();
+        args.putBoolean("is_editable", editable);
+        args.putBoolean("is_featured_image", isFeaturedImage);
+        mediaDetailPagerFragment.setArguments(args);
+        return mediaDetailPagerFragment;
     }
 
+    public MediaDetailPagerFragment() {
+        // Required empty public constructor
+    };
+   
+
```

Comment:
Instead of explicitly declaring a public constructor which does nothing, could just remove it ?

No-arg constructors that do nothing need not be explicitly declared when the class has no other constructors.

Reply:
This piece of code is automatically added to the Fragment when you generate a new Empty Fragment from android Studio with a comment `// Required empty public constructor`. It's maybe to tell others that this empty public constructor is needed or else people do tend to make the constructor private as there is a public `getInstance() `factory method.

Reply:
Ok. The code generation might've been done in the simplest way (since the class may have other constructors too). 

Anyways, this isn't a big deal. Feel free to leave it as-is.